### DISCUSSION TITLE
Inquiry about ROS 2 Version Compatibility in the Master Branch

### DISCUSSION THREAD

**Discussion by animesh-singhal** [cyberbotics/webots_ros2#727] [Q&A] [answered]
Hello everyone,

I hope you're doing well. I recently came across this repository and noticed that the codebase is based on ROS 2. However, I couldn't find any information about which specific version of ROS 2 the master branch utilizes. If any of you have experience with this repository or have knowledge about the supported ROS 2 versions, I kindly request your assistance. It would be immensely helpful to know which ROS 2 version is recommended for use with the master branch and whether there are plans to support other versions in the future.

Best regards, 
Animesh Singhal

**Comment by ygoumaz** [comment, marked as answer]
The `master` branch is compatible with Rolling and Humble, and most examples should work on Foxy as well. However, Navigation and SLAM packages (such as E-puck, TIAGo, and Turtlebot) are no longer supported on Foxy.

To simplify our workflow, we now use only two main branches: `master` and `develop`, instead of one per ROS 2 distribution. Preprocessor directives are used to enable different behavior across ROS 2 distributions. Changes targeting the current version of Webots are committed to `master`, while changes targeting the next version of Webots are committed to `develop`.

**Reply by animesh-singhal** [reply]
Thank you Yannick for your response!
